src/averageAges.js

```diff
@@ -37,7 +51,26 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const countOfWomen = women.filter(isMother => {
+    if (withChildren) {
+      return people.some(person => person.mother === isMother.name);
+    } else {
+      return people.some(person => isMother.name === person.name);
+    }
+  })
+    .reduce(count => (count + 1), 0);
+
+  return women.filter(isMother => {
+    if (withChildren) {
```

Comment:
try use ternary operator here also instead of if:

`withChildren
   ? people.some(person => person.mother === isMother.name)
   : people.some(person => isMother.name === person.name)
)`

Reply:
Bump, why no refactored this code to use the ternary operator :) ? look at my earlier suggestion